src/plugins/gui/fiatProviderTypes.js

```diff
@@ -36,7 +51,7 @@ export type FiatProviderGetQuoteParams = {
 export type FiatProvider = {
   pluginId: string,
   getSupportedAssets: () => Promise<FiatProviderAssetMap>,
-  getQuote: (params: FiatProviderGetQuoteParams) => Promise<FiatProviderQuote | void>
+  getQuote: (params: FiatProviderGetQuoteParams) => Promise<FiatProviderQuote | FiatProviderQuoteError | void>
```

Comment:
Sounds good. I like it. I'll try to rebase those changes but that might be a beast.

Reply:
Oh yeah, the `amount` is not a string because we would never use this as a precise amount that would need to be sent to an exchange. Using a js number makes for each comparison. at the plugin level.

Every other amount is a string because

1. crypto amounts might need full precision if we support Sell and an exchange is expecting perfect accuracy.
2. while fiat amounts never need that much accuracy, it's easier to be consistent with crypto amounts

Reply:
The more I look into it, changing to using Error objects would require using `Promise.allSettled` on the plugin side since we were using `Promise.all` across all calls to `getQuote`. A few issues and inconveniences here:

1. Promise.allSettled isn't well supported on RN. That's ok, it's 4 line function.
2. Promise.allSettled kind of just takes error objects and puts them into a regular object in the returned array of Promise.allSettled. Sort of back to what we were doing.
3. The errors we get in the resulting array will be a mix of FiatQuote errors and other errors caused by a generic throw. So we couldn't really use `error instanceof FiatQuoteLimitError` to check if we got an error we care about so we'd have to check for instanceof every type of FiatQuote error possible. 
4. In the current implementation, the resulting array from getQuote calls has either `undefined` which were timeouts or generic throws, QuoteError objects with a known param we can check (`errorType`), and everything else we know is a good quote.
5. Declaration of the different errors is very verbose: ie

```
export class FiatProviderQuoteLimitError extends Error {
  type: 'overLimit' | 'underLimit'
  amount: number

  constructor(type: 'overLimit' | 'underLimit', amount: number) {
    super('Amount is outside of limits')
    this.type = type
    this.amount = amount
  }
}
```
vs
```
export type FiatProviderQuoteError = { errorType: 'overLimit' | 'underLimit', errorAmount: number }
```

So in summary, while I did give it an honest effort, I'm not sure what benefit we're getting by using Error objects.

Reply:
We did it this way in the swap scene, and it came out nicely. Network failures and other issues force you to handle “real” errors anyhow, and we don’t want misbehaving plugins to lock out working plugins, so while it was more work this way, the reliability of a single code-path was well worth it. All the utilities needed are already in the core.

Reply:
Not sure I understand this. The way it’s coded, no misbehaving plug-ins could cause others to fail. 

Can you give an example of a “real” error that we would have to handle from a plug-in that we don’t do currently? Unless the real errors have an actual error object (and we can always add more) then we don’t have any way to handle them other than to ignore that one plug-in.

Reply:
Also, error objects don't allow us to enforce what type of errors can be thrown. You can enforce return types but not thrown errors. We would require developers to comply with the standard.